marketplace-service/src/main/java/com/axonivy/market/util/MavenUtils.java

```diff
@@ -0,0 +1,292 @@
+    if (StringUtils.isAnyBlank(groupId, artifactId)) {
+      return StringUtils.EMPTY;
+    }
+    repoUrl = Optional.ofNullable(repoUrl).orElse(MavenConstants.DEFAULT_IVY_MAVEN_BASE_URL);
+    groupId = groupId.replace(CommonConstants.DOT_SEPARATOR, CommonConstants.SLASH);
+    return String.join(CommonConstants.SLASH, repoUrl, groupId, artifactId, snapshotVersion,
+        MavenConstants.METADATA_URL_POSTFIX);
+  }
+
+  public static String buildMetadataUrlFromArtifactInfo(String repoUrl, String groupId, String artifactId) {
+    if (StringUtils.isAnyBlank(groupId, artifactId)) {
+      return StringUtils.EMPTY;
+    }
+    repoUrl = Optional.ofNullable(repoUrl).orElse(MavenConstants.DEFAULT_IVY_MAVEN_BASE_URL);
+    groupId = groupId.replace(CommonConstants.DOT_SEPARATOR, CommonConstants.SLASH);
+    return String.join(CommonConstants.SLASH, repoUrl, groupId, artifactId, MavenConstants.METADATA_URL_POSTFIX);
+  }
+
+  public static Metadata convertArtifactToMetadata(String productId, Artifact artifact, String metadataUrl) {
+    String artifactName = artifact.getName();
+    if (StringUtils.isBlank(artifactName)) {
+      artifactName = GitHubUtils.convertArtifactIdToName(artifact.getArtifactId());
+    }
+    String type = StringUtils.defaultIfBlank(artifact.getType(), ProductJsonConstants.DEFAULT_PRODUCT_TYPE);
+    artifactName = String.format(MavenConstants.ARTIFACT_NAME_FORMAT, artifactName, type);
+    return Metadata.builder().groupId(artifact.getGroupId()).versions(new HashSet<>()).productId(productId).artifactId(
+        artifact.getArtifactId()).url(metadataUrl).repoUrl(
+        StringUtils.defaultIfEmpty(artifact.getRepoUrl(), MavenConstants.DEFAULT_IVY_MAVEN_BASE_URL)).type(type).name(
+        artifactName).isProductArtifact(BooleanUtils.isTrue(artifact.getIsProductArtifact())).build();
+  }
+
+  public static Metadata buildSnapShotMetadataFromVersion(Metadata metadata, String version) {
+    String snapshotMetadataUrl = buildSnapshotMetadataUrlFromArtifactInfo(metadata.getRepoUrl(), metadata.getGroupId(),
+        metadata.getArtifactId(), version);
+    return Metadata.builder().url(snapshotMetadataUrl).repoUrl(metadata.getRepoUrl()).groupId(
+        metadata.getGroupId()).artifactId(metadata.getArtifactId()).type(metadata.getType()).productId(
+        metadata.getProductId()).name(metadata.getName()).isProductArtifact(metadata.isProductArtifact()).build();
+  }
+
+  public static MavenArtifactModel buildMavenArtifactModelFromMetadata(String version, Metadata metadata) {
+    return new MavenArtifactModel(metadata.getName(),
+        buildDownloadUrl(metadata.getArtifactId(), version, metadata.getType(),
+            metadata.getRepoUrl(), metadata.getGroupId(), metadata.getSnapshotVersionValue()),
+        metadata.getArtifactId().contains(metadata.getGroupId()));
+  }
+
+  public static String getMetadataContentFromUrl(String metadataUrl) {
+    try {
+      return restTemplate.getForObject(metadataUrl, String.class);
```

Comment:
@nntthuy-axonivy is using URL.openStream to download a content, but @ntqdinh-axonivy is using restTemplate, both worked but can we just unify to use one of them? to make the code consistent